### PR TITLE
Refactors MultipleStoresIT

### LINKED ISSUE #4903: Refactor MultipleStoresIT
**Is your feature request related to a problem? Please describe.**
MultipleStoresIT should be refactored to function more similarly to how other Fate tests are run (e.g., FateIT) where the two stores are tested in their own separate concrete classes. The existing tests use FateTestRunner, but that doesn't work as nicely here since those function under the assumption of one of each store type, which is not the case with this new test.

**Describe the solution you'd like**
MultipleStoresIT to be an abstract class implemented by two new classes: UserMultipleStoresIT and MetaMultipleStoresIT, and a new interface similar to how FateTestRunner is currently used.

**Describe alternatives you've considered**
The current impl is another alternative, which works but isn't as nice as the above suggestion: it isn't consistent with the structure of the existing Fate tests, and consider this in the current impl:
```
@Test
public void testReserveUnreserve() throws Exception {
    testReserveUnreserve(FateInstanceType.META);
    testReserveUnreserve(FateInstanceType.USER);
}
```
If the first call fails, USER will not be run. Would be nicer to have both run irrespective of order and failures.

**Additional context**
See comment https://github.com/apache/accumulo/pull/4524#discussion_r1702305419
See changes from https://github.com/apache/accumulo/pull/4524

### PR DESCRIPTION
**Pull request by kevinrr888**
Refactors MultipleStoresIT to function more similarly to how other fate tests are run (the two store types are tested in their own separate concrete classes). Created MultipleStoresTestRunner which is very similar to FateTestRunner. MultipleStoresIT is now an abstract class implemented by two new classes MetaMultipleStoresIT and UserMultipleStoresIT.

closes #4903

### CODE REVIEW

File: test/src/main/java/org/apache/accumulo/test/fate/MultipleStoresTestRunner.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ */
+package org.apache.accumulo.test.fate;
+
+import java.util.function.Predicate;
+
+import org.apache.accumulo.core.fate.FateStore;
+import org.apache.accumulo.core.fate.zookeeper.ZooUtil;
+import org.apache.zookeeper.KeeperException;
+
+public interface MultipleStoresTestRunner {
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/fate/MultipleStoresTestRunner.java:27]: Could we remove this interface and move the methods it declares into MultipleStoresIT as abstract methods?  No need to do that, just want to make sure I understand the structure of these changes.  Wondering if this interface has any other purposes.

File: test/src/main/java/org/apache/accumulo/test/fate/meta/MetaMultipleStoresIT.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+public class MetaMultipleStoresIT extends MultipleStoresIT {
+  @TempDir
+  private static File TEMP_DIR;
+  private static ZooKeeperTestingServer SZK;
+  private static ZooReaderWriter ZK;
+  private static String FATE_DIR;
+
+  @BeforeAll
+  public static void setup() throws Exception {
+    SharedMiniClusterBase.startMiniCluster();
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/fate/meta/MetaMultipleStoresIT.java:46]: Why does mini cluster need to be started?

**Review by keith-turner — APPROVED** [review]

File: test/src/main/java/org/apache/accumulo/test/fate/MultipleStoresTestRunner.java (hunk 1 above)
Comment (kevinrr888) [reply test/src/main/java/org/apache/accumulo/test/fate/MultipleStoresTestRunner.java]: yeah, not used anywhere else. Moved into MultipleStoresIT

File: test/src/main/java/org/apache/accumulo/test/fate/meta/MetaMultipleStoresIT.java (hunk 2 above)
Comment (kevinrr888) [reply test/src/main/java/org/apache/accumulo/test/fate/meta/MetaMultipleStoresIT.java]: It does not thanks for catching that. Meta and User shared the same setup before so splitting them up I must have carried this over unnecessarily. Removed it in eb7493477eeb36794b14c784503320e2d98fd1dd

**Review by keith-turner — APPROVED** [review]

**Pull request merged**